controller/konnect/ops/ops_kongupstream.go

```diff
@@ -69,38 +63,10 @@ func updateUpstream(
 		},
 	)
 
-	// TODO: handle already exists
-	// Can't adopt it as it will cause conflicts between the controller
-	// that created that entity and already manages it, hm
 	if errWrap := wrapErrIfKonnectOpFailed(err, UpdateOp, upstream); errWrap != nil {
-		// Upstream update operation returns an SDKError instead of a NotFoundError.
-		var sdkError *sdkkonnecterrs.SDKError
-		if errors.As(errWrap, &sdkError) {
-			switch sdkError.StatusCode {
-			case 404:
-				if err := createUpstream(ctx, sdk, upstream); err != nil {
-					return FailedKonnectOpError[configurationv1alpha1.KongUpstream]{
-						Op:  UpdateOp,
-						Err: err,
-					}
-				}
-				// Create succeeded, createUpstream sets the status so no need to do this here.
-
-				return nil
-			default:
-				return FailedKonnectOpError[configurationv1alpha1.KongUpstream]{
-					Op:  UpdateOp,
-					Err: sdkError,
-				}
-			}
-		}
```

Comment:
Why removing `NotFound` handling code?

Reply:
I've realized while working on those PRs that some entities have `Upsert*` functions (like upstream) while some have `Update*` (like controlplane). 

We can either do it here or create a separate PR to handle that ( the latter might actually be better). WDYT?

Reply:
Ah, I see. Makes sense now. Let's keep it in this PR, but for others, it would be better to have separate one(s).

Reply:
Needed to revert that as tests would need changing. Let's keep the original scope for this PR and handle those unnecessary error handling bits separately. They do not harm currently as they are pretty much noop.